packages/core/src/codecs/octetstream-codec.ts

```diff
@@ -142,7 +163,27 @@ export default class OctetstreamCodec implements ContentCodec {
             throw new Error("Undefined value");
         }
 
-        switch (schema.type) {
+        let dataType = schema.type;
+
+        // check @type property for further information
+        if(schema['@type'] !== undefined) {
```

Comment:
Let's assume that the value sent to the underlying protocol binding implementation follows the standard js byte order. In other words, no need to have the endianness described in the type here.